canonical/traefik-k8s-operator
# Refresh url on access

## Issue
https://github.com/canonical/catalogue-k8s-operator/issues/3


## Solution
Refresh url on access ([example](https://github.com/canonical/traefik-route-k8s-operator/commit/905b1b8897740ab1e7ea136304c137c0edf4cf82#diff-b09930835d29e4f97b9b799ee47942218e84c948f74389576c52586657d8a6a0R163)) by pre-fetching urls from relation data instead of stored state.


## Context
See thread: https://github.com/canonical/catalogue-k8s-operator/pull/4#issuecomment-1326823357


## Testing Instructions
Deploy the bundle and see which URL catalogue has for prom: fqdn or ingress.

```bash
$ ./render_bundle.py local.yaml --traefik ../traefik-k8s-operator/traefik-k8s_ubuntu-20.04-amd64.charm
$ juju deploy ./local.yaml --trust
$ juju ssh --container catalogue catalogue/0 cat /web/config.json | jq '.apps'
```


## Release Notes
Prefetch urls from reldata instead of stored state.
